pilot/pkg/config/kube/crd/client.go

```diff
@@ -70,20 +60,81 @@ type IstioObjectList interface {
 
 // Client is a basic REST client for CRDs implementing config store
 type Client struct {
+	// Map of apiVersion to restClient.
+	clientset map[string]*restClient
+
+	// domainSuffix for the config metadata
+	domainSuffix string
+}
+
+type restClient struct {
```

Comment:
The names `restClient` and `Client` are a bit confusing here since both client's are REST clients. Maybe s/`restClient`/`apiGroupClient` in the short term until we add typed clientsets?

Reply:
I'd keep restClient because they are k8s rest clients, while Client is our wrapper client.

Reply:
The comment on line 61 says `Client` is also a REST client, so the distinction wasn't clear.